69-hid-udev.rules

```diff
@@ -0,0 +1,28 @@
+# This is a sample udev file for HIDAPI devices which lets unprivileged
+# users who are physically present at the system (not remote users) access
+# HID devices.
+
+# If you are using the libusb implementation of hidapi (libusb/hid.c), then
+# use something like the following line, substituting the VID and PID with
+# those of your device. Note that for kernels before 2.6.24, you will need
+# to substitute "usb" with "usb_device". It shouldn't hurt to use two lines
+# (one each way) for compatibility with older systems.
+
+# HIDAPI/libusb
+SUBSYSTEM=="usb", ATTRS{idVendor}=="04d8", ATTRS{idProduct}=="003f", TAG+="uaccess"
```

Comment:
since this is an example, lets have an alternative example, where GROUP + MODE is used instead of 'uaccess', which is commonly known among community

Reply:
That is not a good solution because it requires explaining to (human) users that they need to add their user account to a specially named group which may vary depending on the application or distribution. This is an annoying user support burden. I know from experience. `TAG+="uaccess"` does not have this problem.

Reply:
I'm not saying this is a bad solution. For some scenarios it is a great solution. But other than that there is no other reasons to limit users from available options.

Apparently you don't need to run unit-tests for your software from CI over SSH, using a real device connected to the build machine, where access to the device is expected exclusively from a remote session.
Having access setup for a specific USER/GROUP is a great solution for that case too.

Reply:
Does anyone actually do that?

Reply:
At least one suck user for your service.

Reply:
Okay, I added a comment to the end of the file.